### ISSUE TITLE
Why create the direectory yourself in the Docker file?

### ISSUE THREAD

**Issue by sje30** [benmarwick/1989-excavation-report-Madjebebe#1]
hi Ben,

following your examples in Dockerfile (thanks!).  Why do you need to create the directory (line 16of your Dockerfile) first before doing the git clone rather than letting 'git clone' create it?   I had to do the same and didn't understand why.

I also found my Docker file slightly more readable by adding an ENV variable for the root.  See

https://github.com/sje30/waverepo/blob/master/Dockerfile

As ever, I'm learning lots about this from you, thanks.

**Comment by benmarwick** [comment]
I'm not quite sure why that step of creating the directory is there, I guess it was a workaround for an issue that might have been fixed already (maybe related to https://github.com/docker/docker/issues/9136 or https://github.com/docker/docker/issues/1295), or more likely, some voodoo I copied from someone else. I've still got a lot to learn about docker!

Thanks for the tip about the ENV variable, it's does neaten things up quite a bit. I'll try that with my next one. 

Did you consider making an R package for that waverepo paper? I'm curious about how to decide when a package is a suitable compendium for a research project.

**Comment by sje30** [comment]
> Did you consider making an R package for that waverepo paper? I'm
> curious about how to decide when a package is a suitable compendium
> for a research project.

Hi Ben,

I did consider it when I first started putting the paper together, but I
felt the package would be too large, and thus abuse the package idea (I
recall packages supposedly being a few MB at most... I have about 300 Mb
in the data area).  So, my problem would be that I don't think the
package concept makes sense if there is a lot of data.  Probably better
would be the idea of dynamically downloading it from somewehre like
zenodo.

S

**Comment by benmarwick** [comment]
I see, good point about file size. I think CRAN has a limit of 100 Mb for tar.gz files, which is a good indicator of what's normal for R packages (though I guess you wont submit this to CRAN?)

**Comment by sje30** [comment]
> I see, good point about file size. I think CRAN has a limit of 100 Mb
> for tar.gz files, which is a good indicator of what's normal for R
> packages (though I guess you wont submit this to CRAN?)

yes.  I'm wary of putting any package on CRAN, unless I think it has
broad appeal.

S
